portal-impl/src/com/liferay/portlet/asset/service/impl/AssetCategoryLocalServiceImpl.java

```diff
@@ -557,17 +557,15 @@ public AssetCategory mergeCategories(long fromCategoryId, long toCategoryId)
 	@Indexable(type = IndexableType.REINDEX)
 	@Override
 	public AssetCategory moveCategory(
-			long categoryId, long parentCategoryId, long vocabularyId,
+			AssetCategory category, long parentCategoryId, long vocabularyId,
 			ServiceContext serviceContext)
 		throws PortalException {
```

Comment:
Unless there's a reason to do it, can you avoid adding new API? You can make the new `moveCategory` method private and invoke it from both the original `moveCategory` and `updateCategory`.